File: NotificationHubsSample.WinUsingWebApi (v1)/NotificationHubsSample.WinUsingWebApi (v1).Shared/SettingsHelper.cs
using System;
using Windows.Storage;

namespace NotificationHubsSample
{
    /// <summary>
    /// Define the SettingsHelper.
    /// </summary>
    public class SettingsHelper
    {
        private const string ContainerName = "InstallationContainer";
        private const string KeyName = "InstallationId";

        /// <summary>
        /// Gets the local storage container.
        /// </summary>
        /// <returns>ApplicationDataContainer.</returns>
        private static ApplicationDataContainer GetLocalStorageContainer()
        {
            
            if (!ApplicationData.Current.LocalSettings
                .Containers.ContainsKey(ContainerName))
            {
                ApplicationData.Current.LocalSettings
                    .CreateContainer(ContainerName,
                    ApplicationDataCreateDisposition.Always);
            }
            return ApplicationData.Current.LocalSettings.Containers[ContainerName];
        }

        /// <summary>
        /// Gets the installation identifier.
        /// </summary>
        /// <returns>The System.String.</returns>
        public static string GetInstallationId()
        {
            var container = GetLocalStorageContainer();
            if (!container.Values.ContainsKey(KeyName))
            {
                container.Values[KeyName] = Guid.NewGuid().ToString();
            }
            return (string)container.Values[KeyName];
        }
    }
}

File: src/NotificationHubsSample.Win81UsingWebApi (v1) (VB)/SettingsHelper.vb
Imports Windows.Storage

''' <summary>
''' Define the SettingsHelper.
''' </summary>
Public Class SettingsHelper
    Private Const ContainerName As String = "InstallationContainer"
    Private Const KeyName As String = "InstallationId"

    ''' <summary>
    ''' Gets the local storage container.
    ''' </summary>
    ''' <returns>ApplicationDataContainer.</returns>
    Private Shared Function GetLocalStorageContainer() As ApplicationDataContainer

        If Not ApplicationData.Current.LocalSettings.Containers.ContainsKey(ContainerName) Then
            ApplicationData.Current.LocalSettings.CreateContainer(ContainerName, ApplicationDataCreateDisposition.Always)
        End If
        Return ApplicationData.Current.LocalSettings.Containers(ContainerName)
    End Function

    ''' <summary>
    ''' Gets the installation identifier.
    ''' </summary>
    ''' <returns>The System.String.</returns>
    Public Shared Function GetInstallationId() As String
        Dim container = GetLocalStorageContainer()
        If Not container.Values.ContainsKey(KeyName) Then
            container.Values(KeyName) = Guid.NewGuid().ToString()
        End If
        Return DirectCast(container.Values(KeyName), String)
    End Function
End Class